### ISSUE TITLE
URL Patterns doesn't work

### ISSUE THREAD

**Issue by q8X** [amir1376/ab-download-manager#196]
idk if im doing this right but i did try different URLs with or without wildcard and nothing happened 
![2024-11-16_07-01](https://github.com/user-attachments/assets/a81ab815-38d0-4054-9c53-fba070eb1f29)

**Comment by amir1376** [comment]
Hi there. you should also provide category file types. (`exe apk deb rmp` etc...)

**Comment by q8X** [comment]
well this works but don't u think empty means everything?

**Comment by amir1376** [comment]
> don't u think empty means everything

You are right but since URL-pattern is optional (and also other parameters) there should be a non-optional to be exists!
otherwise when the user adds a new link, the first category will always win (it will accept all filetypes and urls etc...) and gives no chance to other categories to be checked.
but thanks for mentioning I will try to improve its logic later. I'll keep it open then

**Comment by LokeYourC3PH** [comment]
> well this works but don't u think empty means everything? or maybe add a way to select all the types

I haven't tried it yet, but I'd imagine that you could just add " * " for filetype and it could work?

But yeah, if that doesn't work, I would prefer empty to mean "everything", since sometimes I'd like to categorize a certain domain into different folders.

**Comment by q8X** [comment]
> I haven't tried it yet, but I'd imagine that you could just add " * " for filetype and it could work?

nope, doesn't work

> But yeah, if that doesn't work, I would prefer empty to mean "everything", since sometimes I'd like to categorize a certain domain into different folders.

iirc, IDM has an option to select all the type or something like that

**Comment by LokeYourC3PH** [comment]
> > I haven't tried it yet, but I'd imagine that you could just add " * " for filetype and it could work?
> 
> nope, doesn't work

Yeah in that case I agree that selecting nothing means selecting everything, or adding " * " as a filter for selecting everything. That way, we can sort not only file types into specific folders, but also downloads from certain URLs into specific folders, which is something I have rarely seen any Download Manager be capable of but it is so damn useful, especially if you are modding games for example.

**Comment by LokeYourC3PH** [comment]
I am still hoping this will get fixed or added, because this is a real killer feature that, as far as I'm concerned, no other Download Manager has properly implemented and would likely boost the user count of this Download Manager significantly, because downloading by domain/sub-domain without selecting a specific file extension and getting it sorted into a particular folder is super useful and really would boost productivity.

**Comment by q8X** [comment]
any update on this?

**Comment by LokeYourC3PH** [comment]
> any update on this?

Seems unlikely at this point considering that, if they wanted to, they would as it is a very minor change really.

**Comment by LokeYourC3PH** [comment]
~~Hello @amir1376 , I was thinking that since this feature would be such an absolute killer feature and many really want it, would it be possible to pay you for adding it? I myself am not versed enough in Kotlin to make a pull request with the feature, and you are very obviously busy, so would paying for it be possible? ABDM might be the best download manager I've used so far (and I've used them all), but it's really just missing the ability to sort things into folders by URL no matter which file type it is, and having that would make it so that I don't need to keep switching anymore.~~

~~If you are interested, let me know 😊~~

**Comment by LokeYourC3PH** [comment]
Alright, never mind, I spent the past hour learning a bit about Kotlin logic, started looking around to figure out where the categorization logic happens, modified it, built it with JBR (I learned to hate gradlew just like NPM :) ), tested it, confirmed that it works. I will make a pull request.

@q8X I got it working with "*" as the file type. I will publish it to my GitHub fork and make a pull request, let's hope they accept it.

**Comment by LokeYourC3PH** [comment]
Please see PR https://github.com/amir1376/ab-download-manager/pull/681

**Comment by q8X** [comment]
> @q8X I got it working with "*" as the file type. I will publish it to my GitHub fork and make a pull request, let's hope they accept it.

oh my, thanks alot! hopefully he accepts it

**Comment by LokeYourC3PH** [comment]
> > [@q8X](https://github.com/q8X) I got it working with "*" as the file type. I will publish it to my GitHub fork and make a pull request, let's hope they accept it.
> 
> oh my, thanks alot! hopefully he accepts it

I can in the meantime, while we wait for approval, link you a build that contains the change if you wish. All you gotta do is replace your current files of ABDM with my build, that's all. If you want it, I will publish and link to it.

**Comment by q8X** [comment]
i've tested your PR, everything works great! but man, the building process is something else 😅, definitely regret not asking you for build link 😄 

![Image](https://github.com/user-attachments/assets/a95d99a7-e6fe-497d-9def-364f6a6160c8)

**Comment by LokeYourC3PH** [comment]
> i've tested your PR, everything works great! but man, the building process is something else 😅, definitely regret not asking you for build link 😄 
> 
> ![Image](https://github.com/user-attachments/assets/a95d99a7-e6fe-497d-9def-364f6a6160c8)

Yea, no kidding, I had to figure out and find several dependencies that were left unmentioned on the GitHub page and build process instructions. After eventually trying for 2 hours, I managed to get it to work due to the aforementioned lack of mentioning all required dependencies.

**Comment by q8X** [comment]
yea, hope @amir1376 update build process instruction to help others to contribute, the current instruction missing alot of things

**Comment by LokeYourC3PH** [comment]
> yea, hope [@amir1376](https://github.com/amir1376) update build process instruction to help others to contribute, the current instruction missing alot of things

Well, we haven't seen nor heard from Amir for 3 weeks now. He used to push updates every few days on average, and make commits roughly bi-daily. Since then, silence. I'm just going on a hunch here, and I hope I am wrong, but seeing as to where Amir is from as it states on his GitHub profile, it might very well be that ABDM won't be updated by Amir ever again. That is just a mere guess and I hope I am wrong, but it's not completely impossible that ABDM's maintenance, at least by Amir, is dead.

I would give it a month, and if by then there's no response, it's pretty safe to assume that Amir lost his life in the Israel-Iran war.

**Comment by amir1376** [comment]
Hi everyone. sorry for the late response.

This improvement has been added, in the next version you can leave the `Category file types` empty, to accept all file types

**Comment by amir1376** [comment]
> yea, hope [@amir1376](https://github.com/amir1376) update build process instruction to help others to contribute, the current instruction missing alot of things

@q8X I tried to document everything needed to build the project, but if you notice anything missing or unclear, feel free to reach out on Telegram, I'm happy to help. Also, any suggestions to improve the instructions are always welcome!

**Comment by amir1376** [comment]
> > yea, hope [@amir1376](https://github.com/amir1376) update build process instruction to help others to contribute, the current instruction missing alot of things
> 
> Well, we haven't seen nor heard from Amir for 3 weeks now. He used to push updates every few days on average, and make commits roughly bi-daily. Since then, silence. I'm just going on a hunch here, and I hope I am wrong, but seeing as to where Amir is from as it states on his GitHub profile, it might very well be that ABDM won't be updated by Amir ever again. That is just a mere guess and I hope I am wrong, but it's not completely impossible that ABDM's maintenance, at least by Amir, is dead.
> 
> I would give it a month, and if by then there's no response, it's pretty safe to assume that Amir lost his life in the Israel-Iran war.

@LokeYourC3PH I'm still around, but due to the current situation and limited internet access, I haven’t been able to update the project for a while. I’ll try to get back to it as soon as I can.

**Comment by LokeYourC3PH** [comment]
> > > yea, hope [@amir1376](https://github.com/amir1376) update build process instruction to help others to contribute, the current instruction missing alot of things
> > 
> > Well, we haven't seen nor heard from Amir for 3 weeks now. He used to push updates every few days on average, and make commits roughly bi-daily. Since then, silence. I'm just going on a hunch here, and I hope I am wrong, but seeing as to where Amir is from as it states on his GitHub profile, it might very well be that ABDM won't be updated by Amir ever again. That is just a mere guess and I hope I am wrong, but it's not completely impossible that ABDM's maintenance, at least by Amir, is dead.
> > 
> > I would give it a month, and if by then there's no response, it's pretty safe to assume that Amir lost his life in the Israel-Iran war.
> 
> @LokeYourC3PH I'm still around, but due to the current situation and limited internet access, I haven’t been able to update the project for a while. I’ll try to get back to it as soon as I can.

I think it's far more important for you to take care of yourself first and foremost. Just glad you weren't hurt 🙂
